sdk/python/foundation-models/system/finetune/Llama-notebooks/distillation/distillation_math.ipynb

```diff
@@ -0,0 +1,599 @@
+    "\n",
+    "\n",
+    "class NLIHuggingFaceInputDataset(InputDataset):\n",
+    "    \"\"\"\n",
+    "    Loads the HuggingFace dataset\n",
+    "    \"\"\"\n",
+    "\n",
+    "    def __init__(self):\n",
+    "        super().__init__()\n",
+    "\n",
+    "    def load_hf_dataset(\n",
+    "        self,\n",
+    "        dataset_name,\n",
+    "        train_sample_size=10,\n",
+    "        val_sample_size=10,\n",
+    "        test_sample_size=10,\n",
+    "        train_split_name=\"train\",\n",
+    "        val_split_name=\"validation\",\n",
+    "        test_split_name=\"test\",\n",
+    "    ):\n",
+    "        full_dataset = load_dataset(dataset_name)\n",
+    "\n",
+    "        if val_split_name is not None:\n",
+    "            train_data = full_dataset[train_split_name].select(range(train_sample_size))\n",
+    "            val_data = full_dataset[val_split_name].select(range(val_sample_size))\n",
+    "            test_data = full_dataset[test_split_name].select(range(test_sample_size))\n",
+    "        else:\n",
+    "            train_val_data = full_dataset[train_split_name].select(\n",
+    "                range(train_sample_size + val_sample_size)\n",
+    "            )\n",
+    "            train_data = train_val_data.select(range(train_sample_size))\n",
+    "            val_data = train_val_data.select(\n",
+    "                range(train_sample_size, train_sample_size + val_sample_size)\n",
+    "            )\n",
+    "            test_data = full_dataset[test_split_name].select(range(test_sample_size))\n",
+    "\n",
+    "        return train_data, val_data, test_data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# We can define train and test sample sizes here. Validation size is kept same as test sample size\n",
+    "train_sample_size = 378\n",
+    "val_sample_size = 42\n",
+    "\n",
+    "# Sample notebook using the dataset: https://huggingface.co/datasets/ChilleD/MultiArith\n",
+    "dataset_name = \"ChilleD/MultiArith\"\n",
+    "input_dataset = NLIHuggingFaceInputDataset()\n",
+    "\n",
+    "# Note: train_split_name and test_split_name can vary by dataset. They are passed as arguments in load_hf_dataset.\n",
+    "# If val_split_name is None, the below function will split the train set to create the specified sized validation set.\n",
+    "train, val, _ = input_dataset.load_hf_dataset(\n",
+    "    dataset_name=dataset_name,\n",
+    "    train_sample_size=train_sample_size,\n",
+    "    val_sample_size=val_sample_size,\n",
+    "    train_split_name=\"train\",\n",
+    "    val_split_name=None,\n",
+    ")\n",
+    "\n",
+    "print(\"Len of train data sample is \" + str(len(train)))\n",
+    "print(\"Len of validation data sample is \" + str(len(val)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! mkdir -p data"
```

Comment:
This creates 2 directories "-p" and "data". We should change this to "!mkdir data".

Reply:
The -p flag does this: no error if existing, make parent directories as needed. Also when checking locally and running ! ls right after, only data directory is created, no "-p" directory is made.

Reply:
<img width="774" alt="image" src="https://github.com/user-attachments/assets/92e0806f-6dd3-4ecc-825b-1b5a517ace93">